### PR TITLE
fixes bug 1386259 - Make sure raven gets the version (sha)

### PR DESCRIPTION
**Pull request by peterbe**
I don't remember why it's "OK" for `resource_string('socorro', 'socorro_revision.txt')` to fail. 
It's [how it's done `settings/base.py`](https://github.com/mozilla-services/socorro/blob/3600d2b70f29fb31660a5aebb797136d73ee1634/webapp-django/crashstats/settings/base.py#L504-L510)

Also, I don't know why I can no longer install `socorro`.

```
▶ python setup.py install
/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/setuptools/dist.py:294: UserWarning: The version specified ('master') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  "details." % self.metadata.version
running install
running bdist_egg
running egg_info
writing socorro.egg-info/PKG-INFO
writing top-level names to socorro.egg-info/top_level.txt
writing dependency_links to socorro.egg-info/dependency_links.txt
writing entry points to socorro.egg-info/entry_points.txt
reading manifest file 'socorro.egg-info/SOURCES.txt'
writing manifest file 'socorro.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.11-x86_64/egg
running install_lib
running build_py
copying socorro/lib/transform_rules.py -> build/lib/socorro/lib
copying socorro/processor/processor_app.py -> build/lib/socorro/processor
copying socorro/external/es/crashstorage.py -> build/lib/socorro/external/es
copying build/lib/socorro/external/es/crashstorage.py -> build/bdist.macosx-10.11-x86_64/egg/socorro/external/es
copying build/lib/socorro/lib/transform_rules.py -> build/bdist.macosx-10.11-x86_64/egg/socorro/lib
copying build/lib/socorro/processor/processor_app.py -> build/bdist.macosx-10.11-x86_64/egg/socorro/processor
byte-compiling build/bdist.macosx-10.11-x86_64/egg/schemas/__init__.py to __init__.pyc
byte-compiling build/bdist.macosx-10.11-x86_64/egg/schemas/validate_and_test.py to validate_and_test.pyc
byte-compiling build/bdist.macosx-10.11-x86_64/egg/sluggish/__init__.py to __init__.pyc
...
byte-compiling build/bdist.macosx-10.11-x86_64/egg/wsgi/__init__.py to __init__.pyc
byte-compiling build/bdist.macosx-10.11-x86_64/egg/wsgi/collector.py to collector.pyc
byte-compiling build/bdist.macosx-10.11-x86_64/egg/wsgi/middleware.py to middleware.pyc
installing package data to build/bdist.macosx-10.11-x86_64/egg
running install_data
error: can't copy 'JENKINS_BUILD_NUMBER': doesn't exist or not a regular file
```

It's been so long since I "worked" on the setup.py stuff (actually I only did some reviewing) that I can't remember how this was supposed to work. 

Anyway, I've tested the processor and now [the release sha](https://sentry.prod.mozaws.net/operations/socorro-stage/issues/632778/) is in the metadata (see right-hand side).

I actually don't know how to test the transform_rules change I made. I just copied what I did in processor_app over.

### CODE REVIEW

File: socorro/processor/processor_app.py [hunk 1]
```diff
@@ -151,7 +160,10 @@ def _transform(self, crash_id):
                         exceptions = exception
                     else:
                         exceptions = [exception]
-                    client = raven.Client(dsn=self.config.sentry.dsn)
+                    client = raven.Client(
+                        dsn=self.config.sentry.dsn,
+                        release=SOCORRO_REVISION,
+                    )
```
Comment (willkg) [inline socorro/processor/processor_app.py]: The code changes in this file are identical to the ones in `socorro.lib.transform_rules`. I think we want to keep them identical. I also think there's some non-trivial work being done. Seems prudent to make this a function in `socorro.lib` somewhere and then use that function in both places to generate a `raven.Client`.

Comment (peterbe) [inline socorro/processor/processor_app.py]: Fair enough. I'll take a stab.

Comment (peterbe) [inline socorro/processor/processor_app.py]: Done.

**Comment by peterbe** [comment]
Note-to-self; Several of the files I touched in this PR have flake8 violations. Once this lands, tidy these up and attach to https://bugzilla.mozilla.org/show_bug.cgi?id=1377254

**Pull request merged**